src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.h

```diff
@@ -156,10 +273,59 @@ using OMapInnerNodeRef = OMapInnerNode::OMapInnerNodeRef;
 
 struct OMapLeafNode
   : OMapNode,
-    StringKVLeafNodeLayout {
-
+    StringKVLeafNodeLayout,
+    ChildNode<OMapInnerNode, OMapLeafNode, std::string> {
   using OMapLeafNodeRef = TCachedExtentRef<OMapLeafNode>;
-  using internal_iterator_t = const_iterator;
+  using internal_const_iterator_t = const_iterator;
+  using base_child_t = BaseChildNode<OMapInnerNode, std::string>;
+  using child_node_t = ChildNode<OMapInnerNode, OMapLeafNode, std::string>;
+
+  void do_on_rewrite(Transaction &t, LogicalCachedExtent &extent) final {
+    auto &ext = static_cast<OMapInnerNode&>(extent);
+    if (ext.is_btree_root()) {
+      set_root();
+    } else {
+      unset_root();
+    }
+    auto &other = static_cast<OMapInnerNode&>(extent);
+    this->init_meta(other.get_begin(), other.get_end());
+  }
+
+  void on_invalidated(Transaction &t) final {
+    this->child_node_t::on_invalidated();
+  }
+
+  void prepare_commit() final {
+    if (is_rewrite() && !is_btree_root()) {
+      auto &prior = *get_prior_instance()->template cast<OMapLeafNode>();
+      if (prior.base_child_t::has_parent_tracker()) {
+	// unlike fixed-kv nodes, rewriting child nodes of the omap tree
+	// won't affect parent nodes, so we have to manually take prior
+	// instances' parent trackers here.
+	this->child_node_t::take_parent_from_prior();
```

Comment:
Reuse `this->child_node_t::on_replace_prior()` and drop `take_parent_from_prior()` ?

Reply:
I thought `on_replace_prior()` may contain other operations in the future that should not be done when committing.